src/ucp/tag/tag_recv.c

```diff
@@ -17,7 +17,7 @@
 static UCS_F_ALWAYS_INLINE ucs_status_t
 ucp_tag_search_unexp(ucp_worker_h worker, void *buffer, size_t count,
                      ucp_datatype_t datatype, ucp_tag_t tag, uint64_t tag_mask,
-                     ucp_request_t *req, ucp_tag_recv_info_t *info)
+                     ucp_request_t *req, ucp_tag_recv_info_t *info, unsigned *save_rreq)
```

Comment:
the second case splits to 2 cases: 
1. when there was a match and need to keep the recv_req (for eager) 
2. when there was a match and don't need to keep the recv_req (for rndv)

so ucp_tag_search_unexp() has to take care of both these cases and the one where there was no match at all. that's why i added the flag..
can add a third return status but that would complicate the functions that call ucp_tag_search_unexp()